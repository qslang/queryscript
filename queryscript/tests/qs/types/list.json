{"a": [1,2,3]}
{"a": []}
{"a": [4,5]}
