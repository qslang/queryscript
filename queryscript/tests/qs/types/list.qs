let lists = load('list.json');

select * from lists;

let foo = 1;
[10, foo];

[];
