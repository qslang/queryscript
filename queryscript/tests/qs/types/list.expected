{
    "compile_errors": [],
    "decls": {
        "let foo": Int64,
        "let lists": External<[{
        	a [UInt64],
        }]>,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "a",
                                type_: List(
                                    Atom(
                                        UInt64,
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| a         |\n|-----------|\n| [1, 2, 3] |\n| []        |\n| [4, 5]    |",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Atom(
                        Int64,
                    ),
                ),
                value: "[10, 1]",
                viz: None,
            },
        ),
        Err(
            TypesystemError {
                source: ArrowError {
                    source: CastError(
                        "Casting from Int32 to Null not supported",
                    ),
                    backtrace: None,
                },
            },
        ),
    ],
}